pkg/audit/newverifier_test.go

```diff
@@ -4,19 +4,313 @@
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		err := planet.Satellites[0].Audit.Service.Close()
+		require.NoError(t, err)
+
+		uplink := planet.Uplinks[0]
+		testData := make([]byte, 8*memory.KiB)
+		_, err = rand.Read(testData)
+		require.NoError(t, err)
+
+		err = uplink.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		projects, err := planet.Satellites[0].DB.Console().Projects().GetAll(ctx)
+		require.NoError(t, err)
+
+		bucketID := []byte(storj.JoinPaths(projects[0].ID.String(), "testbucket"))
+
+		cursor := audit.NewCursor(planet.Satellites[0].Metainfo.Service)
+		stripe, _, err := cursor.NextStripe(ctx)
+		require.NoError(t, err)
+
+		// replace the piece id of the selected stripe with a new random one
+		// to simulate missing piece on the storage nodes
+		stripe.Segment.GetRemote().RootPieceId = storj.NewPieceID()
+
+		verifier := audit.NewVerifier(zap.L(),
+			audit.NewReporter(planet.Satellites[0].Overlay.Service, planet.Satellites[0].DB.Containment(), 1),
+			planet.Satellites[0].Transport,
+			planet.Satellites[0].Overlay.Service,
+			planet.Satellites[0].DB.Containment(),
+			planet.Satellites[0].Orders.Service,
+			planet.Satellites[0].Identity,
+			128*memory.B)
+
+		shareSize := stripe.Segment.GetRemote().GetRedundancy().GetErasureShareSize()
+		limits, err := planet.Satellites[0].Orders.Service.CreateAuditOrderLimits(ctx, planet.Satellites[0].Identity.PeerIdentity(), bucketID, stripe.Segment, nil)
+		require.NoError(t, err)
+
+		shares, _, err := verifier.DownloadShares(ctx, limits, stripe.Index, shareSize)
+		require.NoError(t, err)
+
+		for _, share := range shares {
+			assert.Error(t, share.Error)
+			assert.False(t, transport.Error.Has(share.Error))
+		}
+	})
+}
+
+func TestDownloadSharesDialTimeout(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		err := planet.Satellites[0].Audit.Service.Close()
+		require.NoError(t, err)
+
+		upl := planet.Uplinks[0]
+		testData := make([]byte, 8*memory.KiB)
+		_, err = rand.Read(testData)
+		require.NoError(t, err)
+
+		err = upl.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		projects, err := planet.Satellites[0].DB.Console().Projects().GetAll(ctx)
+		require.NoError(t, err)
+
+		bucketID := []byte(storj.JoinPaths(projects[0].ID.String(), "testbucket"))
+
+		cursor := audit.NewCursor(planet.Satellites[0].Metainfo.Service)
+		stripe, _, err := cursor.NextStripe(ctx)
+		require.NoError(t, err)
+
+		network := &transport.SimulatedNetwork{
+			DialLatency:    200 * time.Second,
+			BytesPerSecond: 1 * memory.KiB,
+		}
+
+		slowClient := network.NewClient(planet.Satellites[0].Transport)
```

Comment:
this test will run for ~1s instead of ~10s if you give the transport client a configurable dial timeout:
```
		tlsOpts, err := tlsopts.NewOptions(planet.Satellites[0].Identity, tlsopts.Config{})
		require.NoError(t, err)

		newTransport := transport.NewClientWithTimeouts(tlsOpts, transport.Timeouts{
			Dial: 20 * time.Millisecond,
		})

		slowClient := network.NewClient(newTransport)
		require.NotNil(t, slowClient)
```
however for `TestDownloadSharesDownloadTimeout`, the `maxTransferTime` is hard coded to default to 5 seconds in the verifier code, so that will need to be changed in order for that test's run time to be reduced.

Reply:
This is an awesome hint! Thank you!